processing/src/main/java/org/apache/druid/query/aggregation/ExpressionLambdaAggregatorFactory.java

```diff
@@ -0,0 +1,536 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.query.aggregation;
+
+import com.fasterxml.jackson.annotation.JacksonInject;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Supplier;
+import com.google.common.base.Suppliers;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.common.guava.Comparators;
+import org.apache.druid.math.expr.Expr;
+import org.apache.druid.math.expr.ExprEval;
+import org.apache.druid.math.expr.ExprMacroTable;
+import org.apache.druid.math.expr.ExprType;
+import org.apache.druid.math.expr.Parser;
+import org.apache.druid.query.expression.ExprUtils;
+import org.apache.druid.segment.ColumnInspector;
+import org.apache.druid.segment.ColumnSelectorFactory;
+import org.apache.druid.segment.column.ColumnCapabilities;
+import org.apache.druid.segment.column.ColumnCapabilitiesImpl;
+import org.apache.druid.segment.column.ValueType;
+import org.apache.druid.segment.virtual.ExpressionPlan;
+import org.apache.druid.segment.virtual.ExpressionPlanner;
+import org.apache.druid.segment.virtual.ExpressionSelectors;
+
+import javax.annotation.Nullable;
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+
+public class ExpressionLambdaAggregatorFactory extends AggregatorFactory
+{
+  private static final String DEFAULT_ACCUMULATOR_ID = "__acc";
+  private static final int DEFAULT_MAX_SIZE_BYTES = 1 << 13;
+
+  private final String name;
+  @Nullable
+  private final Set<String> fields;
+  private final String accumulatorId;
+  private final String foldExpressionString;
+  private final String initialValueExpressionString;
+  private final String initialCombineValueExpressionString;
+
+  private final String combineExpressionString;
+  @Nullable
+  private final String compareExpressionString;
+  @Nullable
+  private final String finalizeExpressionString;
+  private final ExprMacroTable macroTable;
+  private final Supplier<ExprEval<?>> initialValue;
+  private final Supplier<ExprEval<?>> initialCombineValue;
+  private final Supplier<Expr> foldExpression;
+  private final Supplier<Expr> combineExpression;
+  private final Supplier<Expr> compareExpression;
+  private final Supplier<Expr> finalizeExpression;
+  private final int maxSizeBytes;
```

Comment:
Please document that it's ignored when combinedValue is a primitive type.

Reply:
I think it might be better to have a precondition that the size is at minimum the size required to hold a long or double then we wouldn't really have to ignore it because it would be illegal